src/components/routes/homepage/EventFormContainer.js

```diff
@@ -0,0 +1,129 @@
+import React, { useState, forwardRef } from 'react';
+import moment from 'moment';
+import {
+  AppBar,
+  Dialog,
+  Fab,
+  IconButton,
+  Tooltip,
+  Toolbar,
+  Typography,
+  Slide,
+  makeStyles
+} from '@material-ui/core';
+import { Add as AddIcon, Close as CloseIcon } from '@material-ui/icons';
+import { useMutation } from '@apollo/react-hooks';
+import { GET_ALL_EVENTS, CREATE_EVENT } from '../../../gql/events';
+import { EventForm } from './EventForm';
+import { Modal } from './Modal';
+import { Loading } from '../../shared/Loading';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    position: 'fixed',
+    bottom: 20,
+    right: 20
+  },
+  appBar: {
+    backgroundColor: theme.palette.primary.main
+  },
+  title: {
+    marginLeft: theme.spacing(2),
+    flex: 1
+  }
+}));
+
+const Transition = forwardRef(function Transition(props, ref) {
+  return <Slide direction="up" ref={ref} {...props} />;
+});
+
+export const EventFormContainer = () => {
+  const [open, setOpen] = useState(false);
+  const [modal, setModal] = useState({
+    isOpen: false,
+    title: ''
+  });
+
+  const classes = useStyles();
+
+  const [createEvent, { loading, error }] = useMutation(CREATE_EVENT, {
+    refetchQueries: [{ query: GET_ALL_EVENTS }]
+  });
+
+  if (error) {
+    console.log(error);
+  }
+
+  const doCreateEvent = async ({
+    title,
+    venue,
+    start,
+    end,
+    description,
+    image_url: image_base64
+  }) => {
+    await createEvent({
+      variables: {
+        title,
+        venue,
+        start: moment(start),
+        end: moment(end),
+        description,
+        image_base64
+      }
+    })
+      .then(res => {
+        setModal({
+          isOpen: true,
+          title: 'Your event has been created successfully!'
+        });
+      })
+      .catch(err => {
+        setModal({
+          isOpen: true,
+          title: 'An error has occured.'
+        });
+      });
+  };
+
+  return (
+    <>
+      <Modal
+        modalDetails={modal}
+        onClose={() => window.location.replace('/')}
```

Comment:
can't we use `react-router-dom`'s `Link` / `Redirect` element? manually messing up `window.location` might screw up browser history, I feel

Reply:
we could use `history.push('/')` which is being used quiet often in owl

Reply:
I suppose `history` seems fine. Returning a `<Redirect />` here is quite weird tbh

Reply:
yeahh 😂, plus `<Redirect />` is for `BrowserRouter` right?? but we are using `HashRouter` so that doesnt come with it I guess

Reply:
@98sean98 so `history.push('/')` doesnt work since the form component and the events component are on the same route. so doing that will make the page stays on `'/'` but with the latest state (modal being open, eventForm being open). What do you say to `window.location.reload()`

Reply:
lets pause for a moment here. what is this `Modal` trying to do `onClose`? refresh the web page at `/` url?

Reply:
yesss it is, go back to the homepage (`'/'`) with initial states condition

Reply:
in that case, perhaps `window.location.reload()` seems necessary, feels better than `window.location.replace()` not that synthetically it makes a difference though, both reload web pages from scratch

Reply:
or maybe the better ux is to do `setState` on all of those states conditions to initial values...